crates/orderbook/src/main.rs

```diff
@@ -345,10 +336,19 @@ async fn main() {
     if let Some(contract) = &vault {
         finders.push(Arc::new(BalancerVaultFinder(contract.clone())));
     }
+    let uniswapv3_factory = match IUniswapV3Factory::deployed(&web3).await {
+        Err(DeployError::NotFound(_)) => None,
+        other => Some(other.unwrap()),
+    };
+    if let Some(contract) = uniswapv3_factory {
+        finders.push(Arc::new(UniswapV3Finder {
+            factory: contract,
+            base_tokens: base_tokens.tokens().iter().copied().collect(),
+        }));
+    }
```

Comment:
nit: This looks unusual. Since you are only handling `NotFound` and no other error, I assume `other => Some(other.unwrap())` is supposed to throw on every error except `NotFound`.
Could this be made more explicit with:
```rust
match IUniswapV3Factory::deployed(&web3).await {
    Ok(factory) => finders.push(Arc::new(UniswapV3Finder {
        factory,
        base_tokens: base_tokens.tokens().iter().copied().collect(),
    })),
    Err(DeployError::NotFound(_)) => (),
    Err(e) => panic!("{}", e.to_string()),
};
```